code/__HELPERS/colors.dm

```diff
@@ -48,9 +48,36 @@
 
 /// Given a color in the format of "#RRGGBB", will return if the color
 /// is dark.
-/proc/is_color_dark(color, threshold = 25)
-	var/hsl = rgb2num(color, COLORSPACE_HSL)
-	return hsl[3] < threshold
+/proc/is_color_dark(hex_string, threshold = 25, method=BRIGHTNESS_BYOND_COLOR)
+	return get_color_brightness_from_hex(hex_string, method) < threshold
+
+/// returns HSV brightness 0 to 100 by color hex
+/proc/get_color_brightness_from_hex(hex_string, method=BRIGHTNESS_BYOND_COLOR)
```

Comment:
Why do we need different methods? I don't care if you change how existing things work since if they are broken and you are making new procs to fix it then just change it. Switching different methods is the same as making new functions. I don't see why there has to be at all a difference in how we are checking these.

Reply:
There are 2 color check system in the code currently.
* what Kat implemented. (AKA "kat color check")
* weird color checking in the spray code (AKA "bad color check")

What I was requested to do was to change "Bad color check" to "Kat color check" but it's what I don't agree. Kat color check isn't wrong and it's working as intended, but It's not still what I think it's correct to replace "Bad color check", and I've said the reasons above.
The only way to fix it here I believe is to calculate it differently rather than "Kat color check", and it's "Dragon color check" which doesn't count saturation to brightness.
That's why I am putting two methods for checking colors.

Reply:
There should be 2 different procs then, `is_color_dark` and `is_color_unsaturated`. Is colour dark should only check if a colour visually looks dark, which is colours that have a low bright ness and low saturation.

Reply:
This is not getting saturation value. This is why I made an unused proc `get_color_saturation_from_hex()` because this is not a proc to get saturation value.

Byond color `rgb2num(hex_string, COLORSPACE_HSL)` counts saturation + brightness values to conclude if it's dark.
My new proc made it away from saturation value.

I know what you're talking about, but I think I should likely make the whole old `is_color_dark()` procs renamed to "is_color_dark_saturated()"